torch/optim/lr_scheduler.py

```diff
@@ -1319,6 +1321,15 @@ def get_lr(self):
 
         return lrs
 
+    def state_dict(self):
+        # We are dropping the `_scale_fn_ref` attribute because it is a `weakref.WeakMethod` and can't be pickled
+        return {key: value for key, value in self.__dict__.items() if key not in ("optimizer", "_scale_fn_ref")}
+
+    def load_state_dict(self, state_dict):
+        self.__dict__.update(state_dict)
```

Comment:
`super().load_state_dict(state_dict)`

Reply:
Much better! Thanks for the comments, I addressed all of them.